File: C2009i/AP/24-09-2021/SalaryApp/SalaryApp/Models/Manager.cs
using System;
namespace Aptech.Models
{
    public class Manager : Employee, ITax
    {
        public double Bonus { get; set; }
        public void CalculateTax()
        {
            Bonus = Salary * 0.2;
            double tax = Salary * 0.1;
            Salary += Bonus - tax;                
        }
        public override void Display()
        {
            base.Display();
            Console.WriteLine($"Bonus = {Bonus}, Salary = {Salary}");
        }
    }
}

File: C2108G1/ADF1/04-10-2022/studentapp/src/main/java/com/aptech/Manager.java
package com.aptech;

import java.util.Scanner;

public class Manager extends Employee implements Tax{
    private double bonus;
    public Manager(){}
    public Manager(String name, Integer age, int salary, double bonus) {
        super(name, age, salary);
        this.bonus = bonus;
    }

    @Override
    public double calculateTax() {
        bonus = getSalary()*0.2;
        double tax = getSalary()*0.1;
        setSalary((int)(getSalary() + bonus - tax));
        return tax;
    }
    @Override
    public void input() {
        super.input();
        System.out.println("Input bonus:");
        this.bonus = (new Scanner(System.in)).nextDouble();
    }
    @Override
    public void display() {
        super.display();//"employee".display()
        System.out.println("bonus: "+this.bonus);
    }
    @Override
    public String toString() {
        return super.toString()+
                ",bonus:"+bonus;
    }
}
